docs/en/ingest-management/processors/processors.asciidoc

```diff
@@ -0,0 +1,52 @@
+[[elastic-agent-processor-configuration]]
+= Define processors
+
+//REVIEWERS: I think we need to define "processors." Here's my first shot.
+//I think we also need to make it clear that these are different from the
+//processors defined in ingest pipelines. Should we qualify the name?
+
+Processors are lightweight processing components that you can use to parse,
+filter, transform, and enrich data at the source. For example, you can use
+processors to:
+
+* reduce the number of exported fields
+* enhance events with additional metadata
+* perform additional processing and decoding
+* sanitize data
+
+Each processor receives an event, applies a defined action to the event, and
+returns the event. If you define a list of processors, they are executed in the
+order they are defined.
+
+[source,yaml]
+----
+event -> processor 1 -> event1 -> processor 2 -> event2 ...
+----
+
+[discrete]
+[[where-valid]]
+== Where are processors valid?
+
+The processors described in this section are valid:
+
+* Under integration settings in the Integrations UI in {kib}.
+* Under input configuration settings for standalone {agent}s. You can define
+processors that apply to all inputs defined in the configuration, or specific
+inputs.
+
+[discrete]
+[[how-different]]
+== How are {agent} processors different from {ls} plugins or ingest pipelines?
+
+Logstash plugins and ingest pipelines both require you to send data to another
+system for processing. Processors, on the other hand, allow you to apply
+processing logic at the source. This means that you can filter out data you
+don't want to send across the connection, and you can spread some of the
+processing load across host systems running on edge nodes.
```

Comment:
It'd be helpful to bullet list some of the use cases that can't be solved with agent processors and other limitations:
- Can't enrich with data from ES or other custom data sources
- Can't process data after it's been converted to ECS (since that's done in ES pipelines)
- User-defined processing could break integration ingest pipelines in ES. I don't know much about how defensive we are about preventing this in our integrations, but it's certainly possible. Would be good to discuss with integration devs about how likely this is and an example we could show. @andrewkroh any thoughts?

Reply:
Added this list. I'll keep this issue open in case andrew has anything to add.